redisearch/aggregate.go

```diff
@@ -0,0 +1,179 @@
+package redisearch
+
+import "github.com/garyburd/redigo/redis"
+
+// Projection
+type Projection struct {
+	Expression string
+	Alias      string
+}
+
+func NewProjection(expression string, alias string) *Projection {
+	return &Projection{
+		Expression: expression,
+		Alias:      alias,
+	}
+}
+
+func (p Projection) Serialize() redis.Args {
+	args := redis.Args{"APPLY", p.Expression, "AS", p.Alias}
+	return args
+}
+
+// GroupBy
+type GroupBy struct {
+	Fields   []string
+	Reducers []Reducer
+	Paging   *Paging
+}
+
+func NewGroupBy(field string) *GroupBy {
```

Comment:
Why not creating an empty GROUPBY and then expose a function to add fields?

Reply:
done. please revise the new commit. Do you like AddFields in the following manner? It prevents us from having to create two nearly equal methods. 
```
func (g *GroupBy) AddFields(fields interface{}) *GroupBy
```